vm/src/main/scala/pravda/vm/standard/ExponentialFunction.scala

```diff
@@ -0,0 +1,317 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package pravda.vm.standard
+
+import pravda.vm.Data.Primitive.{BigInt, Int16, Int32, Int8, Number, Uint16, Uint32, Uint8}
+import pravda.vm.Data.Type
+import pravda.vm.VmError.WrongType
+import pravda.vm.WattCounter.CpuArithmetic
+import pravda.vm._
+
+object ExponentialFunction extends FunctionDefinition {
+  override def id: Long = 0x03L
+
+  override def description: String = "Calculate exponential operation first two items in the stack and push result into"
+
+  override def args: Seq[(String, Seq[Type])] = Seq(
+    "x" -> Seq(Data.Type.Int32),
+    "y" -> Seq(Data.Type.Int32)
+  )
+
+  override val returns = Seq(Data.Type.Number)
+  private val div: Double = math.log10(2.0)
+  override def apply(memory: Memory, wattCounter: WattCounter): Unit = {
+    implicit class DoubleExp(x: Double) {
+      def **(y: Double): Double = math.pow(x, y)
+      def **(y: Byte): Double = math.pow(x, y.toDouble)
+      def **(y: Int): Double = math.pow(x, y.toDouble)
+      def **(y: Long): Double = math.pow(x, y.toDouble)
+      def **(y: Short): Double = math.pow(x, y.toDouble)
+    }
+    implicit class LongExp(x: Long) {
+      def **(y: Double): Long = math.pow(x.toDouble, y).toLong
+      def **(y: Byte): Long = math.pow(x.toDouble, y.toDouble).toLong
+      def **(y: Int): Long = math.pow(x.toDouble, y.toDouble).toLong
+      def **(y: Long): Long = math.pow(x.toDouble, y.toDouble).toLong
+      def **(y: Short): Long = math.pow(x.toDouble, y.toDouble).toLong
+    }
+    implicit class IntExp(x: Int) {
+      def **(y: Double): Int = math.pow(x.toDouble, y).toInt
+      def **(y: Byte): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Int): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Long): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Short): Int = math.pow(x.toDouble, y.toDouble).toInt
+    }
+    implicit class ShortExp(x: Short) {
+      def **(y: Double): Short = math.pow(x.toDouble, y).toShort
+      def **(y: Byte): Short = math.pow(x.toDouble, y.toDouble).toShort
+      def **(y: Int): Short = math.pow(x.toDouble, y.toDouble).toShort
+      def **(y: Long): Short = math.pow(x.toDouble, y.toDouble).toShort
+      def **(y: Short): Short = math.pow(x.toDouble, y.toDouble).toShort
+    }
+    implicit class ByteExp(x: Byte) {
+      def **(y: Double): Int = math.pow(x.toDouble, y).toInt
+      def **(y: Byte): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Int): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Long): Int = math.pow(x.toDouble, y.toDouble).toInt
+      def **(y: Short): Int = math.pow(x.toDouble, y.toDouble).toInt
+    }
+
+    def getCost(x: Double, y: Double): Long = {
+      (math.log10(math.abs(x) + 1) / div * math.abs(y) / 8).toLong + 1
+    }
+
+    def getBigIntCost(x: scala.BigInt, y: Int): Long = {
+      (x.abs.toString().length / div * math.abs(y) / 8).toLong + 1
```

Comment:
`if (true.toString.length == 4) { ... }`

Take a look at `.bitCount`, `.bitLength` methods. I'd also simplify this method in terms of constants.